Fix types

ATTW isn't running anymore, presumably I need to fix the types again.

I'll probably do this as part of #286.

I can confirm things are broken. I just spun up a new project and can't run eslint..

Steps to reproduce:
* Spin up a new project.
* run...
```sh
 npm init @sherifforg/config
Need to install the following packages:
@sherifforg/create-config@9.1.0
Ok to proceed? (y) 
                                                                                                                                                              8:53:59 PM
Welcome to Sheriff! 👋                                                                                                                                        8:53:59 PM
This wizard will guide you through the setup of Sheriff in your project.                                                                                      8:53:59 PM
Let's get started!                                                                                                                                            8:53:59 PM

✔ Do you want to use the eslint-ts-patch?
Yes
◐ No ESLint config files were found. Generating and configuring eslint.config.ts file...                                                                      8:54:02 PM

 WARN  If you have other ESLint configs in your project, remove them                                                                                          8:54:02 PM

◐ 'Astro' package found in the project. Setting up support for it...                                                                                          8:54:02 PM
ℹ Setting Sheriff with options:                                                                                                                              8:54:02 PM

 ╭──────────────────────────────────────────────╮
 │                                              │
 │  {                                           │
 │    "react": false,                           │
 │    "lodash": false,                          │
 │    "remeda": false,                          │
 │    "next": false,                            │
 │    "astro": true,                            │
 │    "playwright": false,                      │
 │    "jest": false,                            │
 │    "vitest": false                           │
 │  }                                           │
 │                                              │
 ╰──────────────────────────────────────────────╯

✔ Successfully generated eslint.config.ts file                                                                                                               8:54:03 PM
◐ No 'prettier' configuration or dependency was found in the project. Generating and configuring .prettierrc.json file...                                     8:54:03 PM
✔ Successfully generated .prettierrc.json file                                                                                                               8:54:03 PM
◐ No .prettierignore file was found in the project. Generating and configuring .prettierignore file...                                                        8:54:03 PM
◐ Installing 'eslint'...                                                                                                                                      8:54:03 PM
◐ Installing 'eslint-define-config'...                                                                                                                        8:54:03 PM
◐ Installing 'eslint-config-sheriff'...                                                                                                                       8:54:03 PM
◐ Installing 'eslint-ts-patch'...                                                                                                                             8:54:03 PM
✔ Successfully generated .prettierignore file                                                                                                                8:54:03 PM
ℹ Detected package manager: npm                                                                                                                              8:54:03 PM
◐ Installing dependendencies...                                                                                                                               8:54:03 PM
npm warn ERESOLVE overriding peer dependency
npm warn ERESOLVE overriding peer dependency
npm warn ERESOLVE overriding peer dependency
npm warn ERESOLVE overriding peer dependency
npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated glob@7.1.7: Glob versions prior to v9 are no longer supported

added 387 packages, removed 2 packages, changed 19 packages, and audited 822 packages in 1m

328 packages are looking for funding
  run `npm fund` for details

8 vulnerabilities (1 low, 2 moderate, 5 high)

To address all issues, run:
  npm audit fix

Run `npm audit` for details.
✔ eslint and eslint-define-config and eslint-config-sheriff and eslint-ts-patch and eslint@npm:eslint-ts-patch were installed successfully                   8:55:30 PM
ℹ You're all set!                                                                                                                                            8:55:30 PM
➜  workspace git:(master) ✗ npm run lint

> astro@0.0.1 lint
> eslint --fix .


Oops! Something went wrong! :(

ESLint: 9.15.0

ESLint couldn't find an eslint.config.(js|mjs|cjs) file.

From ESLint v9.0.0, the default configuration file is now eslint.config.js.
If you are using a .eslintrc.* file, please follow the migration guide
to update your configuration file to the new format:

https://eslint.org/docs/latest/use/configure/migration-guide

If you still have problems after following the migration guide, please stop by
https://eslint.org/chat/help to chat with the team.

➜  workspace git:(master) ✗ npm run lint

> astro@0.0.1 lint
> eslint --flag unstable_ts_config --fix .


/project/workspace/eslint.config.ts
  1:8  error  Using exported name 'sheriff' as identifier for default import  import/no-named-as-default

✖ 1 problem (1 error, 0 warnings)
```

The installer creates this `eslint.congig.ts`:
```ts
import sheriff, { type SheriffSettings } from 'eslint-config-sheriff';
import { defineFlatConfig } from 'eslint-define-config';

const sheriffOptions: SheriffSettings = {
  "react": false,
  "lodash": false,
  "remeda": false,
  "next": false,
  "astro": true,
  "playwright": false,
  "jest": false,
  "vitest": false
};

export default defineFlatConfig([...sheriff(sheriffOptions)]);
```

However changing it to this fixes it:
```diff
+++ b/eslint.config.ts
@@ -1,15 +1,15 @@
-import sheriff, { type SheriffSettings } from 'eslint-config-sheriff';
-import { defineFlatConfig } from 'eslint-define-config';
+import sheriffConfig, { type SheriffSettings } from "eslint-config-sheriff";
+import { defineFlatConfig } from "eslint-define-config";
 
 const sheriffOptions: SheriffSettings = {
-  "react": false,
-  "lodash": false,
-  "remeda": false,
-  "next": false,
-  "astro": true,
-  "playwright": false,
-  "jest": false,
-  "vitest": false
+  react: false,
+  lodash: false,
+  remeda: false,
+  next: false,
+  astro: true,
+  playwright: false,
+  jest: false,
+  vitest: false,
 };
 
-export default defineFlatConfig([...sheriff(sheriffOptions)]);
+export default defineFlatConfig([...sheriffConfig(sheriffOptions)]);
(END)
```

--- a/.github/workflows/merge-checks.yml
+++ b/.github/workflows/merge-checks.yml
@@ -10,7 +10,13 @@ on:
 
 jobs:
   merge-checks:
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+
+    runs-on: ${{ matrix.os }}
     if: '!github.event.pull_request.draft'
     steps:
       - name: Checkout

--- a/.github/workflows/merge-checks.yml
+++ b/.github/workflows/merge-checks.yml
@@ -30,3 +30,5 @@ jobs:
         uses: rharkor/caching-for-turbo@v1.5
       - name: Run qa checks
         run: pnpm merge-checks
+      - name: Check deduped deps
+        run: pnpm exec turbo check-deduped-deps

--- a/.github/workflows/merge-checks.yml
+++ b/.github/workflows/merge-checks.yml
@@ -36,5 +36,3 @@ jobs:
         uses: rharkor/caching-for-turbo@v1.5
       - name: Run qa checks
         run: pnpm merge-checks
-      - name: Check deduped deps
-        run: pnpm exec turbo check-deduped-deps

--- a/package.json
+++ b/package.json
@@ -19,7 +19,7 @@
     "typecheck": "turbo run typecheck",
     "knip": "knip",
     "validate-config": "turbo run validate-config",
-    "merge-checks": "pnpm build && turbo run publint manypkg typesync knip typecheck lint validate-config check-deduped-deps are-the-types-wrong",
+    "merge-checks": "pnpm build && turbo run publint manypkg typesync knip typecheck lint validate-config are-the-types-wrong",
     "check-deduped-deps": "pnpm dedupe --check",
     "clean": "turbo run clean && rm -rf .turbo",
     "delete-node-modules": "pnpm exec rm -rf node_modules && pnpm -r exec rm -rf node_modules",

--- a/package.json
+++ b/package.json
@@ -19,7 +19,7 @@
     "typecheck": "turbo run typecheck",
     "knip": "knip",
     "validate-config": "turbo run validate-config",
-    "merge-checks": "pnpm build && turbo run publint manypkg typesync knip typecheck lint validate-config are-the-types-wrong",
+    "merge-checks": "pnpm build && turbo run publint manypkg typesync knip typecheck lint validate-config are-the-types-wrong && turbo run check-deduped-deps",
     "check-deduped-deps": "pnpm dedupe --check",
     "clean": "turbo run clean && rm -rf .turbo",
     "delete-node-modules": "pnpm exec rm -rf node_modules && pnpm -r exec rm -rf node_modules",

--- a/packages/eslint-config-sheriff/dts-config.json
+++ b/packages/eslint-config-sheriff/dts-config.json
@@ -1,0 +1,12 @@
+{
+  "entries": [
+    {
+      "filePath": "./dist/index.d.ts",
+      "outFile": "./dist/index.d.ts",
+
+      "libraries": {
+        "inlinedLibraries": ["@sherifforg/types", "@sherifforg/constants"]
+      }
+    }
+  ]
+}

--- a/packages/eslint-config-sheriff/package.json
+++ b/packages/eslint-config-sheriff/package.json
@@ -46,7 +46,7 @@
   "scripts": {
     "preinstall": "npx only-allow pnpm",
     "clean": "rm -rf .turbo dist node_modules/.cache",
-    "re-bundle-types": "dts-bundle-generator -out-file=dist/index.d.ts ./dist/index.d.ts",
+    "re-bundle-types": "dts-bundle-generator --config dts-config.json",
     "build": "tsup && pnpm re-bundle-types",
     "build-watch": "tsup && pnpm re-bundle-types",
     "typecheck": "tsc --noEmit",

--- a/packages/eslint-config-sheriff/package.json
+++ b/packages/eslint-config-sheriff/package.json
@@ -47,6 +47,7 @@
     "preinstall": "npx only-allow pnpm",
     "clean": "rm -rf .turbo dist node_modules/.cache",
     "re-bundle-types": "dts-bundle-generator --config dts-config.json",
+    "are-the-types-wrong": "attw --pack --profile=esm-only",
     "build": "tsup && pnpm re-bundle-types",
     "build-watch": "tsup && pnpm re-bundle-types",
     "typecheck": "tsc --noEmit",
@@ -55,6 +56,7 @@
     "typesync": "typesync --dry=fail"
   },
   "dependencies": {
+    "@arethetypeswrong/cli": "^0.17.2",
     "@eslint-react/eslint-plugin": "^1.10.1",
     "@eslint/compat": "^1.2.4",
     "@eslint/js": "^9.17.0",

--- a/packages/eslint-config-sheriff/tsup.config.js
+++ b/packages/eslint-config-sheriff/tsup.config.js
@@ -9,4 +9,11 @@ export default defineConfig({
   noExternal: ['@sherifforg/constants', '@sherifforg/types'],
   skipNodeModulesBundle: true,
   dts: { entry: 'src/index.ts' },
+  // banner: ({ format }) => {
+  //   return format === 'esm'
+  //     ? {
+  //         js: `import { createRequire as __createRequire } from 'node:module'; const require = __createRequire(import.meta.url);`,
+  //       }
+  //     : {};
+  // },
 });

--- a/packages/eslint-config-sheriff/tsup.config.js
+++ b/packages/eslint-config-sheriff/tsup.config.js
@@ -8,15 +8,5 @@ export default defineConfig({
   clean: true,
   noExternal: ['@sherifforg/constants', '@sherifforg/types'],
   skipNodeModulesBundle: true,
-  dts: {
-    entry: 'src/index.ts',
-    resolve: true,
-  },
-  banner: ({ format }) => {
-    if (format === 'esm')
-      return {
-        js: `import { createRequire } from 'module'; const require = createRequire(import.meta.url);`,
-      };
-    return {};
-  },
+  dts: { entry: 'src/index.ts' },
 });

--- a/turbo.json
+++ b/turbo.json
@@ -49,7 +49,7 @@
       "persistent": true,
       "dependsOn": ["build"]
     },
-    "check-deduped-deps": {
+    "//#check-deduped-deps": {
       "inputs": [
         "package.json",
         "apps/*/package.json",

